### PR TITLE
Add phpseclib as a replace entry in composer.json

### LINKED ISSUE #21: Compatibility with PHP 7.x.x.
**FILE: google-site-kit/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP/Stream.php**

> ----------------------------------------------------------------------------------------------------------------------------
> FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
> ----------------------------------------------------------------------------------------------------------------------------
>  182 | WARNING | Global with anything other than bare variables is discouraged since PHP 7.0. Found ${$host}
> ----------------------------------------------------------------------------------------------------------------------------

**FILE: google-site-kit/vendor/phpseclib/phpseclib/phpseclib/bootstrap.php**

> ----------------------------------------------------------------------------------------------------------------------
> FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
> ----------------------------------------------------------------------------------------------------------------------
>  10 | WARNING | INI directive 'mbstring.func_overload' is deprecated since PHP 7.2
> ----------------------------------------------------------------------------------------------------------------------
> 

**FILE: google-site-kit/vendor/phpseclib/phpseclib/phpseclib/Crypt/Base.php**

> ----------------------------------------------------------------------------------------------------------------------------------------------------------
> FOUND 45 ERRORS AND 1 WARNING AFFECTING 26 LINES
> ----------------------------------------------------------------------------------------------------------------------------------------------------------
>   792 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   792 | ERROR   | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>   825 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   825 | ERROR   | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>   828 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   828 | ERROR   | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>   833 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   833 | ERROR   | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>   842 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   842 | ERROR   | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>   852 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   852 | ERROR   | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>   855 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>   855 | ERROR   | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1132 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1132 | ERROR   | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1160 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1160 | ERROR   | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1165 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1165 | ERROR   | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1174 | ERROR   | Function mdecrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1177 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1177 | ERROR   | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1659 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1659 | ERROR   | Function mcrypt_list_algorithms() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1738 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1738 | ERROR   | Function mcrypt_module_close() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1739 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1739 | ERROR   | Function mcrypt_module_close() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1744 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1744 | ERROR   | Function mcrypt_module_close() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1851 | ERROR   | The constant "MCRYPT_MODE_ECB" is deprecated since PHP 7.1 and removed since PHP 7.2
>  1852 | ERROR   | The constant "MCRYPT_MODE_CBC" is deprecated since PHP 7.1 and removed since PHP 7.2
>  1854 | ERROR   | The constant "MCRYPT_MODE_CFB" is deprecated since PHP 7.1 and removed since PHP 7.2
>  1855 | ERROR   | The constant "MCRYPT_MODE_NOFB" is deprecated since PHP 7.1 and removed since PHP 7.2
>  1856 | ERROR   | The constant "MCRYPT_MODE_STREAM" is deprecated since PHP 7.1 and removed since PHP 7.2
>  1859 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1859 | ERROR   | Function mcrypt_module_open() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1860 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1860 | ERROR   | Function mcrypt_module_open() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1866 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1866 | ERROR   | Function mcrypt_module_open() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  1866 | ERROR   | The constant "MCRYPT_MODE_ECB" is deprecated since PHP 7.1 and removed since PHP 7.2
>  1871 | ERROR   | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
>  1871 | ERROR   | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
>  2610 | WARNING | Function create_function() is deprecated since PHP 7.2; Use an anonymous function instead
> ----------------------------------------------------------------------------------------------------------------------------------------------------------

**FILE: google-site-kit/vendor/phpseclib/phpseclib/phpseclib/Crypt/Random.php**

-------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 2 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------
  69 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  69 | ERROR | Function mcrypt_create_iv() is deprecated since PHP 7.1 and removed since PHP 7.2; Use random_bytes() or OpenSSL instead
 108 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 108 | ERROR | Function mcrypt_create_iv() is deprecated since PHP 7.1 and removed since PHP 7.2; Use random_bytes() or OpenSSL instead
 108 | ERROR | The constant "MCRYPT_DEV_URANDOM" is deprecated since PHP 7.1 and removed since PHP 7.2
-------------------------------------------------------------------------------------------------------------------------------------------------------

**FILE: google-site-kit/vendor/symfony/yaml/Parser.php**

> --------------------------------------------------------------------------------------------------
> FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
> --------------------------------------------------------------------------------------------------
>  72 | WARNING | INI directive 'mbstring.func_overload' is deprecated since PHP 7.2
> --------------------------------------------------------------------------------------------------

**FILE: google-site-kit/vendor/symfony/yaml/Inline.php**

> --------------------------------------------------------------------------------------------------
> FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
> --------------------------------------------------------------------------------------------------
>  55 | WARNING | INI directive 'mbstring.func_overload' is deprecated since PHP 7.2
> --------------------------------------------------------------------------------------------------
> 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Changelog entry

* 

## Acceptance criteria

*

### PR DESCRIPTION
**Pull request by ataylorme**
## Summary

Addresses issue #21

## Relevant technical choices
Inspired by #43 but removing the unwanted dependency via Composer so that the autoloader is not problematic.

The `phpseclib/phpseclib` dependency of `google/apiclient` is not PHP 7+ compatible and not used by Site Kit, so it should be removed. Since it is an inherited dependency the removal needs to be done via Composer.

The `composer.json` can have `replace` entries to specify dependencies fulfilled by the current project, which will not be installed by Composer. The intention is to use forked or self-maintained versions of a dependency but, in this case, can be used to skip the dependency entirely. See [the `composer.json` replace doc](https://getcomposer.org/doc/04-schema.md#replace) for details.

## Checklist:
- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,5 +36,8 @@
   "scripts": {
     "lint": "vendor/bin/phpcs",
     "lint-fix": "vendor/bin/phpcbf"
+  },
+  "replace": {
+    "phpseclib/phpseclib": "*"
```
Comment (felixarntz) [inline composer.json]: Minor request: Could you move this below the `require` and `require-dev` directives?

**Comment by felixarntz** [comment]
cc @schlessera Would be great to have your feedback on this.

File: composer.json (hunk 1 above)
Comment (ataylorme) [reply composer.json]: Updated in [`439e7f1`](https://github.com/google/site-kit-wp/pull/72/commits/439e7f126a5d8b8a063b8a7e2f2137ea28f92395)

**Comment by ataylorme** [comment]
> Thanks for the PR! I was looking into a solution, but now you beat me to it, and it works great. :)

It is a little hacky but I don't see another way with the dependency you want removed being managed by Composer. A better solution would be an upstream change to `phpseclib/phpseclib` for PHP 7 compatibility. 

> Only one thing, can you also incorporate the change from #43 into this PR please?

Done in [`4b3cdcd`](https://github.com/google/site-kit-wp/pull/72/commits/4b3cdcdf38316c439cd0dcbec3ad31c9a49a1b83)

**Comment by schlessera** [comment]
Yes, looking good.

Apart from changing `phpseclib/phpseclib` to be PHP 7 compatible, another way to solve the issue would be to change `google/apiclient` to make the `phpseclib/phpseclib` optional (behind a safeguard) and move it from `"require"` to `"suggest"`.

**Pull request merged**